gradio/flagging.py

```diff
@@ -248,6 +248,20 @@ def setup(self, components: list[IOComponent], flagging_dir: str):
             repo_type="dataset",
             exist_ok=True,
         ).repo_id
+        path_glob = "**/*.jsonl" if self.separate_dirs else "data.csv"
```

Comment:
By default, dataset preview and datasets library will load all json files. This is a problem because we serialize some components to json, e.g. ChatBot and Gallery. This is so that only the jsonl files are used to compute the features. Thanks @severo and @ albertvillanova for the help.

Reply:
Nice